Split thermodynamic_coefficients_type

mnlevy1981 opened the pull request:
Per #19, have put dic, ta, pt, and sit in their own data type.

I used the name `thermodynamic_species_concentration_type` based on the comment above the declaration (`declarations for function coefficients & species concentrations`) but am happy to change it to something more appropriate.

Verified this code builds on the usual machines / compilers, and that it does not change answers in my one-off ciso gx3v7 test on hobart (nag) or cheyenne (intel).

mnlevy1981 committed:
Split thermodynamic_coefficients_type

Per #19, have put dic, ta, pt, and sit in their own data type.

mnlevy1981 commented:
For this PR, I was uncertain about what to name the new derived type -- does the `thermodynamic` prefix in `thermodynamic_species_concentration_type` make sense? I'm happy to change it, and intentionally didn't change the whitespace in variable declaration blocks in case the `species_concentration` line shrinks.

klindsay28 commented:
Given that the name of the module contains co2calc, I suggest renaming
thermodynamic_coefficients_type to co2calc_coeffs_type
thermodynamic_coefficients_type to co2calc_state_type

I see an oppurtunity for the related renamings to make names more consistent
co3_coeffs to co2calc_coeffs
marbl_comp_co3_coeffs to comp_co2calc_coeffs (also list this as private at the module level)

mnlevy1981 committed:
More thermo_coeff cleanup following review

thermodynamic_coefficients_type          -> co2calc_coeffs_type
thermodynamic_species_concentration_type -> co2calc_state_type

Next commit will have more of the recommended cleanup:
* co3_coeffs -> co2calc_coeffs
* better subroutine names in module

mnlevy1981 committed:
Merge branch 'master' into enhancement/cleanup_thermo_coeff_type

mnlevy1981 committed:
co3_coeffs -> co2calc_coeffs and more code cleanup

Per the second suggestion of the code review, co3_coeffs has been
renamed co2calc_coeffs (and species_concentration has been renamed
co2calc_state to match the derived type name). marbl_comp_co3_coeffs has
been renamed comp_co2calc_coeffs, and explicitly declared as private in
the module. And I believe white space in variable declarations / code
continuation lines is all lined up correctly.

mnlevy1981 commented:
@klindsay28 -- can you take a look at this PR again when you have a chance? I believe I've addressed everything in your comment. Thanks!

klindsay28 reviewed (approved):
looks good
